app/v1/controllers/requestParsers/validators/validations/IncomeSourceIdIncludedWithTypeOfBusinessValidation.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers.validators.validations
+
+import v1.models.errors.{MissingTypeOfBusinessError, MtdError}
+
+object IncomeSourceIdIncludedWithTypeOfBusinessValidation {
+
+  def validate(businessId: Option[String], typeOfBusiness: Option[String]): List[MtdError] = {
+    (businessId, typeOfBusiness) match {
+      case (None, Some(x)) => List(MissingTypeOfBusinessError)
```

Comment:
Wrong way around - this error is when there is a incomeSourceId but no typeOfBusiness.